src/transformers/modeling_utils.py

```diff
@@ -2340,6 +2340,11 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
                 load_in_8bit=load_in_8bit,
             )
 
+        # Let's add a safety checker to warn users that `save_pretrained` cannot be used
+        # under 8_bit models
+        if load_in_8bit:
+            setattr(model, "save_pretrained", warn_uncompatible_save(model.save_pretrained))
```

Comment:
Mmm, let's store an attribute `is_laoded_in_8bit` instead of wrapping the `save_pretrained` method.

Reply:
Thanks for the suggestion! Proposed an alternative in bc955d4